packages/core/src/actions/invoke.ts

```diff
@@ -43,8 +43,12 @@ export function invoke<
         } as InvokeActionObject;
       }
 
+      const behaviorImpl = machine.options.actors[src.type];
       const behaviorCreator: BehaviorCreator<TContext, TEvent> | undefined =
-        machine.options.actors[src.type];
+        behaviorImpl &&
+        (typeof behaviorImpl === 'function'
+          ? behaviorImpl
+          : () => behaviorImpl);
```

Comment:
nit: in general it would be better here to just conditionally resolve the `behaviorImpl` at the line 60 instead of creating a temporary, throwaway, function here when the `behaviorImpl` is not a function (and not undefined)

Reply:
I've created a PR to address this: https://github.com/statelyai/xstate/pull/3281